### DISCUSSION TITLE
Environmental variable not making its way through to polybar

### DISCUSSION THREAD

**Discussion by kurtlieber** [polybar/polybar#3023] [Q&A] [answered]
I'm trying to add host-specific polybar configurations via the include-file feature, but for some reason, the environmental variables I"m setting aren't getting through to polybar. The relevant part from my launcher script:

```
HOST_FILE="${HOME}/.config/polybar/$(hostname).config"

if type "xrandr" > /dev/null; then
  for m in $(xrandr --query | grep " connected" | cut -d" " -f1); do
    if [ $m == 'eDP-1' ]
    then
        MONITOR=$m HOST_FILE="${HOST_FILE}" polybar --reload main -c ~/.config/polybar/config &
    elif [ $m == 'DP-1-3' ]
    then
        MONITOR=$m HOST_FILE="${HOST_FILE}" polybar --reload external-dp-1-3 -c ~/.config/polybar/config &
    elif [ $m == 'DP-1' ]
    then
        MONITOR=$m HOST_FILE="${HOST_FILE}" polybar --reload external-dp-1 -c ~/.config/polybar/config &
    elif [ $m == 'HDMI-1' ]
    then
        MONITOR=$m HOST_FILE="${HOST_FILE}" polybar --reload external-hdmi-1 -c ~/.config/polybar/config &
#    else
#       MONITOR=$m HOST_FILE="${HOST_FILE}" polybar --reload sidebar-2-i3 -c ~/.config/polybar/config &
    fi
  done
else
    polybar --reload main -c ~/.config/polybar/config &
fi

```

And the include-file from my polybar config:

`include-file = ${env:HOST_FILE}`

Running that produces:

```

klieber@archibald:>./launch.sh

polybar-msg: No active ipc channels

notice: Parsing config file: /home/klieber/.config/polybar/config

notice: Parsing config file: /home/klieber/.config/polybar/config

error: Uncaught exception, shutting down: Config file /home/klieber/.config/polybar// is a directory

error: Uncaught exception, shutting down: Config file /home/klieber/.config/polybar// is a directory

```

(I have two monitors which is why the message repeats twice)

If I change nothing else except the include-file directive to:

`include-file = /home/klieber/.config/polybar/archibald.config`

Then everything works as expected.

I have verified that $HOST_FILE is being properly set in the launch script via echo statements - it's just not being passed through to polybar for some reason.

What am I doing wrong?

**Comment by patrick96** [comment, marked as answer]
`include-file` is handled differently than other settings, because it's not really a setting, more of directive.

It does not have the same resolution mechanisms as regular settings. But maybe we should still support env variables.

As a workaround you could do `include-file = ~/.config/polybar/config` in `archibald.config` and start polybar with `polybar -c "$HOST_FILE"`

**Reply by kurtlieber** [reply]
Thank you - that explains it and I will use the workaround suggested.  Appreciate the help.
